src/main/java/com/google/plantasy/utils/UserUtils.java

```diff
@@ -163,20 +165,22 @@ public static boolean addBlobKey(String blobKey, Entity userEntity, DatastoreSer
         return false;
     }
     
-    userEntity.setProperty("blobKey", blobKey);
-    datastore.put(userEntity);
- 
+    User user = new User(userEntity);
+    user.setBlobKey(blobKey);
+
     return true; 
   }
  
   /**
     adds a specified number of points to the user's points
   */
   public static void addPoints(Entity userEntity, int numPoints, DatastoreService datastore){
+    User user = new User(userEntity);
+
     try {
-      userEntity.setProperty("score", ((Number) userEntity.getProperty("score")).intValue() + numPoints);
+      user.setScore(user.getScore() + numPoints);
     } catch (NullPointerException e) {
-      userEntity.setProperty("score", numPoints);                
+      user.setScore(numPoints);                
     }
     datastore.put(userEntity);
```

Comment:
I think this needs to be changed from     datastore.put(userEntity);	to     datastore.put(user.GetUserEntity());	

Might want to check anywhere else we do datastore.put and make sure we're using the wrapper class and not the old entity.

Reply:
Got it, that's done in all the places with datastore.put()